src/components/playback/playbackmanager.js

```diff
@@ -1888,7 +1888,10 @@ class PlaybackManager {
 
             if (options.items) {
                 return translateItemsForPlayback(options.items, options).then(function (items) {
-                    return playWithIntros(items, options);
+                    getAdditionalParts(items).then(function (allItems) {
+                        const flattened = allItems.flatMap(i => i);
+                        return playWithIntros(flattened, options);
+                    });
```

Comment:
Not tested, but there will be less nested Promises this way.
```
return translateItemsForPlayback(options.items, options)
    .then(getAdditionalParts)
    .then(function (allItems) {
        const flattened = allItems.flatMap(i => i);
        return playWithIntros(flattened, options);
    });
```

Reply:
Done